lib/post_destroyer.rb

```diff
@@ -91,6 +91,7 @@ def recover
       user_recovered
     end
     topic = Topic.with_deleted.find @post.topic_id
+    topic.update_column(:user_id, Discourse.system_user.id) if !topic.user_id
```

Comment:
Even thought we're caching the system user, I think it might be faster to only use the constant here?

```suggestion
    topic.update_column(:user_id, Discourse::SYSTEM_USER_ID) if !topic.user_id
```